### PR TITLE
declare conflict after moving headers

### LINKED ISSUE #2648: file conflict when upgrading moveit_ros_planning
### Description

Updating `moveit_ros_planning` (a.k.a. `ros-noetic-moveit-ros-planning`) from 1.1.0 to 1.1.2 breaks apt.

### Your environment
- Ubuntu 20.04, Noetic
- `ros-noetic-moveit-ros-planning` `1.1.0-1focal.20210302.142257`

### Steps to reproduce
I am not sure if this is reproducible.
1. install `ros-noetic-moveit-ros-planning` and `ros-noetic-moveit-ros-planning-interface` version `1.1.0-1focal.20210302`
2. `sudo apt upgrade` to version `1.1.2-1focal.20210424`

### Expected behaviour
Upgrade without issues.

### Actual behaviour
apt fails and interrupts all system updates

### Backtrace or Console output
```
Unpacking ros-noetic-moveit-ros-planning (1.1.2-1focal.20210424.003930) over (1.1.0-1focal.20210302.142257) ...
dpkg: error processing archive /tmp/apt-dpkg-install-Mi1pqr/127-ros-noetic-moveit-ros-planning_1.1.2-1focal.20210424.003930_amd64.deb (--unpack):
 trying to overwrite '/opt/ros/noetic/include/moveit/moveit_cpp/moveit_cpp.h', which is also in package ros-noetic-moveit-ros-planning-interface 1.1.0-1focal.20210302.150335
dpkg-deb: error: paste subprocess was killed by signal (Broken pipe)
```

This is probably because file `/opt/ros/noetic/include/moveit/moveit_cpp/moveit_cpp.h` moved between packages. But if packages are not upgraded in the correct order, this will lead to conflicts. Unfortunately, this will interrupt all system updates until resolved manually.

MoveIt has the tendency to break API/ABI during stable releases, but please don't break the system update.

### PR DESCRIPTION
**Pull request by v4hn**
... that install to the same location.

Following the suggestion from https://github.com/ros-planning/moveit/issues/2648#issuecomment-836647351

Fixup for eb3bf6fdb6ece5324e3d1a81fd5d2b2197804710

Fixes #2648.

### Description

Please explain the changes you made, including a reference to the related issue if applicable

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by v4hn** [comment]
Assuming this works as expected (hard to test...), we should definitely push out a 1.1.4 release before the next sync.
Or maybe actually get this line as a fixup into 1.1.2 which is rolled out for some time now.

File: moveit_ros/planning/package.xml [hunk 1]
```diff
@@ -40,6 +40,9 @@
   <test_depend>moveit_resources_panda_moveit_config</test_depend>
   <test_depend>rostest</test_depend>
 
+  <!-- we moved moveit/moveit_cpp/... from there after this version, so files conflict -->
+  <conflict version_lte="1.1.0">moveit_ros_planning_interface</conflict>
```
Comment (rhaschke) [inline moveit_ros/planning/package.xml:44]: #2127 was merged between 1.1.1 and 1.1.2.
```suggestion
  <conflict version_lte="1.1.1">moveit_ros_planning_interface</conflict>
```

**Review by rhaschke — COMMENTED** [review]
I think the mentioned release number is wrong. Otherwise looks good to me. No idea, how to test this, though.

**Comment by jspricke** [comment]
You probably want Breaks/Replaces, see: https://wiki.debian.org/PackageTransition

**Comment by gavanderhoorn** [comment]
> This is because the buildfarm will always append a `-` version to the package.

Bloom adds that increment, during the run of the `rosdebian` generator.

**Comment by v4hn** [comment]
> Bloom adds that increment, during the run of the `rosdebian` generator.

Jochen does not want to reply any further here. :D
The increment is added there, yes. The bigger problem is the build data though, which is apparently added by the buildfarm.

**Comment by v4hn** [comment]
> instead of posting this here in a comment with zero visibility, reporting this upstream may be more productive.
> It's unlikely to immediately be addressed, but at least it'd be tracked somewhere more appropriate.

It's enough visibility to get you involved. ;)
I agree but spent most of my work time with MoveIt related issues today already.
If you know where to report this, please feel free to copy-paste/reference my comment there.

This PR was reviewed by multiple competent people so I'll go ahead and merge.
We definitely need a quick release to get this into the ROS repositories.

File: moveit_ros/planning/package.xml [hunk 2]
```diff
@@ -40,6 +40,9 @@
   <test_depend>moveit_resources_panda_moveit_config</test_depend>
   <test_depend>rostest</test_depend>
 
+  <!-- we moved moveit/moveit_cpp/... from there after this version, so files conflict -->
+  <conflict version_lte="1.1.1">moveit_ros_planning_interface</conflict>
```
[the remark this answers is not in the document]
Comment (rhaschke) [reply moveit_ros/planning/package.xml]: I don't think `replace` is correct here! `moveit_ros_planning` doesn't replace `moveit_ros_planning_interface`. Both packages stay in place. We just need to enforce a specific purge order.

[the remark this answers is not in the document]
Comment (jspricke) [reply moveit_ros/planning/package.xml]: well, it is what Debian policy argues to use. Note that we are in apt land here, not in ROS or C++ land so replace means something different.

[the remark this answers is not in the document]
Comment (rhaschke) [reply moveit_ros/planning/package.xml]: Ok. I will trust you on this ;-)

**Comment by v4hn** [comment]
> Ok. I will trust you on this ;-)

We discussed this exact point for roughly an hour this morning. :)
Despite the name the tag does not require a *full* replacement, a single replaced path is enough.

File: moveit_ros/planning/package.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jspricke) [reply moveit_ros/planning/package.xml]: just for reference:
"If the breaking package also overwrites some files from the older package, it should use Replaces to ensure this goes smoothly"
https://www.debian.org/doc/debian-policy/ch-relationships.html#packages-which-break-other-packages-breaks

[the remark this answers is not in the document]
Comment (rhaschke) [reply moveit_ros/planning/package.xml]: Thanks for the reference!

**Pull request merged**